test/onnx/internal/fx/test_frontend_consistency.py

```diff
@@ -0,0 +1,157 @@
+
+_FrontendType = Union[frontend.FxFrontend, frontend.FxFrontendUnpackKwargs]
+FX_FRONTENDS: Sequence[_FrontendType] = [
+    # TODO(bowbao): skip aotautograd, because it alters graph inputs/outputs.
+    # frontend.FxFrontendUnpackKwargs(frontend.AOTAutogradFrontend(dynamic=True)),
+    frontend.DynamoExport(tracing_mode="symbolic", aten_graph=True),
+    frontend.DynamoExport(tracing_mode="real", aten_graph=True),
+    frontend.FxFrontendUnpackKwargs(frontend.DynamoOptimize(dynamic=True)),
+    frontend.FxFrontendUnpackKwargs(frontend.MakeFx(tracing_mode="real")),
+    frontend.FxFrontendUnpackKwargs(frontend.DynamoOptimize(dynamic=False)),
+]
+
+# TODO: Organize test models into a separate file to be re-used by other tests.
+class MLPModel(nn.Module):
+    def __init__(self):
+        super().__init__()
+        self.fc0 = nn.Linear(8, 8, bias=True)
+        self.fc1 = nn.Linear(8, 4, bias=True)
+        self.fc2 = nn.Linear(4, 2, bias=True)
+        self.fc3 = nn.Linear(2, 2, bias=True)
+
+    def forward(self, tensor_x: torch.Tensor):
+        tensor_x = self.fc0(tensor_x)
+        tensor_x = torch.sigmoid(tensor_x)
+        tensor_x = self.fc1(tensor_x)
+        tensor_x = torch.sigmoid(tensor_x)
+        tensor_x = self.fc2(tensor_x)
+        tensor_x = torch.sigmoid(tensor_x)
+        output = self.fc3(tensor_x)
+        return output
+
+
+def module_inputs_mlp_model(
+    module_info, device, dtype, requires_grad, training, **kwargs
+):
+    make_input = functools.partial(
+        common_modules.make_tensor,
+        device=device,
+        dtype=dtype,
+        requires_grad=requires_grad,
+    )
+    module_inputs = [
+        common_modules.ModuleInput(
+            constructor_input=common_modules.FunctionInput(),
+            forward_input=common_modules.FunctionInput(make_input((97, 8))),
+        ),
+    ]
+    return module_inputs
+
+
+MODULES_DB = [
+    common_modules.ModuleInfo(MLPModel, module_inputs_func=module_inputs_mlp_model)
+]
+
+
+class TestFxFrontendConsistency(common_utils.TestCase):
+    @classmethod
+    def create_test_base(cls, fx_frontend: _FrontendType):
+        """Returns the base test method for the given fx frontend."""
+
+        @common_modules.modules(
+            MODULES_DB, train_eval_mode=common_modules.TrainEvalMode.eval_only
+        )
+        def _output_match_base(
+            self: common_utils.TestCase,
+            device: str,
+            dtype: torch.dtype,
+            module_info: common_modules.ModuleInfo,
+            training: common_modules.TrainEvalMode,
+        ):
+            """Base test method for testing each fx frontend, used by instantiate_device_type_tests."""
+            # device is provided by instantiate_device_type_tests, but we only want to run in cpu.
+            assert device == "cpu"
+
+            module_cls = module_info.module_cls
+            module_inputs = module_info.module_inputs_func(
+                module_info,
```

Comment:
`module_inputs_func` is a class method of `module_info`. Why is it called with `module_info`?

Reply:
I guess you meant a ~~class~~ method of `module_info`?

It's actually an attribute, assigned when initializing `ModuleInfo`.